### PR TITLE
beta.2

### PR DESCRIPTION
**Pull request by jordanebelanger**

### CODE REVIEW

File: Sources/Redis/Application+Redis.swift [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+            typealias Value = RedisConfiguration
+        }
+
+        public var configuration: RedisConfiguration? {
+            get {
+                self.application.storage[ConfigurationKey.self]
+            }
+            nonmutating set {
+                if self.application.storage.contains(PoolKey.self) {
+                    fatalError("Cannot modify your Redis configuration after redis has been used")
```
Comment (jordanebelanger) [inline Sources/Redis/Application+Redis.swift]: I think you definitely want to fatalError here and not warning, at least in Redis case, for security reasons.

Imagine you have a client that uses a default redis for most things, but has a special redis for secure and or confidential data.

The user might think he can simply swap the configuration during runtime at random times to write to secure redis, but if we keep this scenario as a warning, a user may end up writing confidential data to the wrong redis. 

By the time the warning is taken into account, the damage may already have been done. Simply ignoring is not an option here.

**Comment by jordanebelanger** [comment]
@tanner0101  I'm also thinking, since we're using a pool, we may want to expose the pool directly. There are certain cases where you may want to send more than one command one after an other, in those cases, doing 2 redis.send in a row will thread lock twice for no reason since you could just be holding the connection for the duration of the 2+ calls instead. 

It would just be `app.redis.pool`

File: Sources/Redis/Application+Redis.swift [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+                return existing
+            } else {
+                let lock = self.application.locks.lock(for: PoolKey.self)
+                lock.lock()
+                defer { lock.unlock() }
+                guard let configuration = self.configuration else {
+                    fatalError("Redis not configured. Use app.redis.configuration = ...")
+                }
+                let new = EventLoopGroupConnectionPool(
+                    source: RedisConnectionSource(configuration: configuration, logger: self.application.logger),
```
Comment (Mordil) [inline Sources/Redis/Application+Redis.swift]: The `logger` in this argument is which `Logger` will handle logs generated by the `RedisConnectionSource` - do we _not_ want to use the [default value](https://github.com/vapor/redis-kit/blob/master/Sources/RedisKit/RedisConnectionSource.swift#L14)?

Comment (tanner0101) [reply Sources/Redis/Application+Redis.swift]: @Mordil application's logger may contain useful context in the future. If there's a situation where > 1 application is running and outputting data (common in testing), having the context could be helpful in debugging.

**Comment by tanner0101** [comment]
> doing 2 redis.send in a row will thread lock twice for no reason since you could just be holding the connection for the duration of the 2+ calls instead.

Hmm why would that thread lock at all?

**Comment by jordanebelanger** [comment]
> > doing 2 redis.send in a row will thread lock twice for no reason since you could just be holding the connection for the duration of the 2+ calls instead.
> 
> Hmm why would that thread lock at all?

Unless I am mistaken, the older ConnectionPool used to synchronize access on "requestConnection" calls with a lock. So if that was indeed the case, 2 or more redis send calls in a row would needlessly lock the pool multiple times to retrieve a connection each time.

But it looks like the new EventLoopConnectionPool doesn't need to synchronize anymore upon a `requestConnection` call so indeed you are right no lock is needed.

**Pull request merged**